src/infra/core/src/ingest/fetch_service/mod.rs

```diff
@@ -235,11 +233,11 @@ impl FetchService {
 
                 if let Some(cenv) = re_env.captures(ctpl.get(1).unwrap().as_str().trim()) {
                     let env_name = cenv.get(1).unwrap().as_str();
+
                     let dataset_env_var_secret_value = self
                         .dataset_key_value_svc
-                        .find_dataset_env_var_value_by_key(env_name, dataset_env_vars)
-                        .await
-                        .int_err()?;
```

Comment:
This service even had an error defined for this case already - `IngestParameterNotFound` and it was regressed into internal error :(